examples/shouldi/shouldi/safety.py

```diff
@@ -29,14 +29,14 @@
         "check",
         "--stdin",
         "--json",
-        stdin=asyncio.subprocess.PIPE,
-        stdout=asyncio.subprocess.PIPE,
-        stderr=asyncio.subprocess.PIPE,
+        stdin=asyncio.subprocess.Process.communicate,
+        stdout=asyncio.subprocess.Process.communicate,
+        stderr=asyncio.subprocess.Process.communicate,
     )
 
-    proc.stdin.write(pinned.encode())
-    proc.stdin.write(b"\n")
-    proc.stdin.close()
+    proc.communicate(pinned.encode())
+    proc.communicate(b"\n")
+    proc.communicate()
 
     stdout, _stderr = await proc.communicate()
```

Comment:
This line should be the only one that needs changing. Check the documentation for [`asyncio.subprocess.Process.communicate`](https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.asyncio.subprocess.Process.communicate), you just need the one call, just change it so that it passes `pinned.encode()` and the newline.